src/pseudopeople/dataset.py

```diff
@@ -0,0 +1,154 @@
+from pathlib import Path
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+from vivarium import ConfigTree
+
+from pseudopeople.configuration import Keys
+from pseudopeople.constants.metadata import DATEFORMATS
+from pseudopeople.constants.noise_type_metadata import COPY_HOUSEHOLD_MEMBER_COLS
+from pseudopeople.entity_types import ColumnNoiseType, NoiseType, RowNoiseType
+from pseudopeople.loader import load_data
+from pseudopeople.schema_entities import COLUMNS, Dataset
+from pseudopeople.utilities import coerce_dtypes, get_randomness_stream
+
+
+class DatasetData:
+    def __init__(
+        self,
+        dataset: Dataset,
+        input_data: Union[Path, pd.DataFrame],
+        user_filters: List[Tuple],
+        seed: Any,
+    ):
+        self.dataset = dataset
+        self.data = load_data(input_data, user_filters)
+        self.randomness = get_randomness_stream(self.dataset.name, seed, self.data.index)
+        self.missingness = self.is_missing(self.data)
+
+    def __bool__(self):
+        return not self.data.empty
+
+    def is_empty(self, column_name: str) -> bool:
+        """Returns whether the column is empty."""
+        return self.missingness[column_name].all()
+
+    def get_non_empty_index(self, required_columns: Optional[List[str]] = None) -> pd.Index:
+        """Returns the non-empty data."""
+
+        if required_columns is None:
+            non_empty_data = self.data.loc[~self.missingness.all(axis=1)]
```

Comment:
Here in the if block, we are checking if all columns are empty or a better way to think about it - are there rows that have no data in any of the columns and if so we do not want to remove those rows so we do not choose them to noise. In the else block, we subset to just the columns needed for a specific column noise and we remove any index where there is any cell with missing data so that index is not selected for noising.

So yes they are handled differently but the main reason is required_columns is not an argument for row noise types.